### ISSUE TITLE
Editor: better controls for default camera

### ISSUE THREAD

**Issue by looeee** [mrdoob/three.js#16008]
Most modelling programs have quick settings for the default camera, such as:

* jump to specific views such as top, left, bottom, right, 
* reset the view
* switch between perspective and orthographic projection
* fit camera to scene (in clara.io this is called "frame scene")
* fit camera to selection
* create new perspective camera from current view and add it to the scene

It would be useful to add these to the editor as well. Most of these options can go in the same menu that was added in #15998.
See Clara.io for an example of how these can be set up.

**Comment by Temdog007** [comment]
These are good ideas. I'm trying not to add to many PRs before they merged (as per @Mugen87 [comment](https://github.com/mrdoob/three.js/pull/15792#issuecomment-464209631)). 

I would suggest waiting until #15750, #15998, and #16010 are merged before implementing these changes. Those PRs add menus that would make adding these features easier.

**Comment by mrdoob** [comment]
#15998 and #16010 merged.  #15750 to go 💪

**Comment by Temdog007** [comment]
Since we're close, I'll start getting some feedback from you guys now about how to tackle these suggestions. I don't want the issue that occurred in #15983 where I was told to basically redo my implementation.
***
> - jump to specific views such as top, left, bottom, right
> - switch between perspective and orthographic projection

These can be done by adding adding an `OrthographicCamera` to the `Editor` class.

Current | Suggestion
----- | -------
<img width="213" alt="cameraSelectionOld" src="https://user-images.githubusercontent.com/37945129/55114901-6b7f3900-50a0-11e9-865c-4f3d05726421.png"> | <img width="210" alt="cameraSelectionNew" src="https://user-images.githubusercontent.com/37945129/55114910-6fab5680-50a0-11e9-97f7-1b125b217e96.png">

The change in #15998 will allow for camera selection.
***
> - reset the view

I'm assuming you mean set the view back to the default position the editor sets the camera on startup. @looeee, please correct me if I'm wrong.

> Most of these options can go in the same menu that was added in #15998.

I was thinking that that menu should be for camera selection only. I was thinking about adding the option in the edit toolbar instead.

<img width="245" alt="cameraReset" src="https://user-images.githubusercontent.com/37945129/55115366-8dc58680-50a1-11e9-9d36-6088516414b2.png">

***
> - fit camera to scene (in clara.io this is called "frame scene")
> - fit camera to selection

Doesn't the editor already do this? If you double click an object, the camera focuses on that object. To focus on the scene, just double click the scene in the `Outliner`. The editor currently only allows for one object to be selected at once. So to select a selection of objects, put those objects into a `Group`, then double click the `Group` in the `Outliner`. Unless, you want the editor to allow for multiple objects to be selected?
***
> - create new perspective camera from current view and add it to the scene

I would just add a context menu to the default editor cameras but only have the 'clone' option available.
<img width="212" alt="cameraClone" src="https://user-images.githubusercontent.com/37945129/55115970-07aa3f80-50a3-11e9-814b-a96b6e3acb3a.png">

#15750 would need to be merged before I work on that one.

**Comment by Mugen87** [comment]
>fit camera to scene (in clara.io this is called "frame scene")
fit camera to selection

Are you aware that such features are already somewhat implemented? If you double-click on entries in the outliner,`THREE.EditorControls` focuses the selected object. That works for individual objects as well as for groups or scenes.

**Comment by looeee** [comment]
Ah, no I didn't know about that. Works well, although I wonder how we can communicate better that it's available? Double-clicking is not a common way of interacting with things on the web.

**Comment by Mugen87** [comment]
We could add a menu entry that triggers this logic?

**Comment by looeee** [comment]
The camera menu in the top right is currently fairly empty, maybe there?
Maybe a shortcut key well? The Z key is used for zoom to selection in Max, I don't know if that's the case in other apps.

**Comment by Mugen87** [comment]
> Maybe a shortcut key well?

This is already possible 😁 . Just select an option and the press `f` (for focus).

**Comment by looeee** [comment]
Yeah, currently the Help menu just links to the source code and the threejs.org homepage, which is not very helpful haha. 

Are the keyboard shortcuts listed anywhere? That would be a good start.

**Comment by Mugen87** [comment]
> Are the keyboard shortcuts listed anywhere?

The are listed in the "SETTINGS"  tab where you can customize them, too.

**Comment by Mugen87** [comment]
> jump to specific views such as top, left, bottom, right,

Implemented via #19774. 🙌

**Comment by mrdoob** [comment]
@Mugen87 I think I would prefer enhancing `EditorControls`.

`EditorControls` was a fork of `OrbitControls` that allowed me adding features (like double click to focus) without having to think of consequences 😅

**Comment by Mugen87** [comment]
> I think I would prefer enhancing EditorControls.

I also prefer this option 😊 . `OrbitControls` is always some sort of a hot topic so having `EditorControls` gives us more freedom within the editor.
